packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/MarkerController.java

```diff
@@ -7,82 +7,139 @@
 import com.google.android.gms.maps.model.BitmapDescriptor;
 import com.google.android.gms.maps.model.LatLng;
 import com.google.android.gms.maps.model.Marker;
+import com.google.maps.android.collections.MarkerManager;
+import java.lang.ref.WeakReference;
 
 /** Controller of a single Marker on the map. */
 class MarkerController implements MarkerOptionsSink {
 
-  private final Marker marker;
+  // Holds a weak reference to a Marker instance. The clustering library
+  // dynamically manages markers, adding and removing them from the map
+  // as needed based on user interaction or data changes.
+  private final WeakReference<Marker> weakMarker;
   private final String googleMapsMarkerId;
   private boolean consumeTapEvents;
 
   MarkerController(Marker marker, boolean consumeTapEvents) {
-    this.marker = marker;
+    this.weakMarker = new WeakReference<>(marker);
```

Comment:
Can you add a comment here explaining why a weakreference is needed? I believe that you would not have added the complexity if it was not needed but it helps when someone else comes to modify the code.

Reply:
Isn't the comment on few lines above is enough explaining the need for the weakReference?
As the clustering process might remove the marker from the map (when clustered), the marker should not be accessed anymore in MarkerController. Instead markerbuilder is updated, so that if the marker is declustered, it is used to update the marker settings before  drawing.

Reply:
yes that comment works for me.